lib/logstash/plugin_mixins/aws_config/v2.rb

```diff
@@ -31,6 +31,10 @@ def aws_options_hash
 
     opts[:endpoint] = @endpoint unless @endpoint.nil?
 
+    if respond_to?(:additional_settings)
+      opts.merge! symbolize_keys_and_cast_true_false(additional_settings)
```

Comment:
should maybe merge the [other way around for compatibility](https://github.com/logstash-plugins/logstash-input-s3/blob/v3.8.3/lib/logstash/inputs/s3.rb#L444) as well as not to override explicit settings:
```
  endpoint => 'foo'
  additional_settings => { endpoint => 'bar' }
```
... would end up with `endpoint => bar`, while the current behavior (in S3) is that it ends up as `endpoint => foo`

Reply:
Good point, fixed with [190bfa5](https://github.com/logstash-plugins/logstash-mixin-aws/pull/53/commits/190bfa5bbafb01516d975fa0de7ae812001ef3d7)

Reply:
@kares if you want it's ready for a second review